packages/core/src/ceramic.ts

```diff
@@ -685,12 +685,10 @@ export class Ceramic implements CeramicApi {
   async multiQuery(queries: Array<MultiQuery>, timeout?: number): Promise<Record<string, Stream>> {
     const queryResults = await Promise.all(
       queries.map((query) => {
-        try {
-          return this._loadLinkedStreams(query, timeout)
-        } catch (e) {
+        return this._loadLinkedStreams(query, timeout).catch(e => {
           this._logger.warn(`Error during multiQuery: ${e.toString()}`)
-          return Promise.resolve({})
-        }
+          return {}
+        })
```

Comment:
Why is this changed?

Reply:
This is the most controversial change of all here. Original code, it seems, contained a bug.

There is a `try-catch` clause around `_loadLinkedStreams` call, which conveys that an error should be caught. Really, as `_loadLinkedStreams` returns a `Promise`, its rejection is not caught by `try-catch` at all without `await` statement. If adding `await`, `try-catch` clause is too verbose and does not directly tell "just log an error, do nothing" story.

Backstory: if left as it were, the unhandled promise rejection weirdly interfered with Jest and go-ipfs. It prematurely ended the test suite: killed go-ipfs instances (as specified in `afterAll` hook), but did not close ceramic instances, as they are down the control flow of the test. I guess, in general it could cause a multiquery to throw if just one of the queries failed.

Reply:
good catch

Reply:
so I think the essence of this change is right, but it has an unfortunate side effect made evident by the changes to ceramic.test.ts.  I think if a user uses the API wrong, they should get an error right away in their app that they are misusing the API.  With this change, now it'll just silently fail.

The problem is that we're treating user-input errors (thrown by `_ensureGenesis`) the same as a failure to load a stream (an error or timeout when `_loadLinkedStreams` loads a stream, which has it's own try-catch).

I think the thing to do would be to move the `_ensureGenesis` call out of `_loadLinkedStreams` and up to `multiQuery`.  So we could then do 2 passes over all the `queries` in `multiQuery` - one pass to `_ensureGenesis` for each, where any failures causes the entire multiquery to throw an error, and then a second pass to `_loadLinkedStreams` where all errors are swallowed.

Reply:
@stbrody This changes current semantics of multiquery: individual queries execute separately. Suddenly, there is no way to return back an individual error here.

Reply:
And, I tend to believe, if we are to handle this erroring behavior, we better approach it holistically. One approach I can think of is to return an error message there in a query entry, but discussion on error message format might take a bit too long.

Reply:
> This changes current semantics of multiquery

I think it's the PR as written that is changing the semantics worse than what I am proposing.  Multiquery currently today will throw an error if the user mis-uses the `genesis` api.  Now this PR removes that error and makes it fail silently instead.  I'm trying to preserve the existing semantics of the application getting an error if they mis-use the API.

It's true that if multiple queries in the list all make the same mistake in how the `genesis` option works, the app will only get an error from the first one.  But this doesn't seem so bad to me, and I agree that figuring out a format for returning and correlating multiple errors from multiple queries is out of scope here.  But chances are that if an application is mis-using the `genesis` api, probably all the queries misuse it in the same way, and just getting an error from the first misuse would cause them to fix their app and likely fix the issue for all the queries in the multiquery.